src/pages/DelegatoryValidator/Components/TimeDurationIntervalBar.tsx

```diff
@@ -12,37 +14,55 @@ export default function TimeDurationIntervalBar({
     return null;
   }
 
+  const [remainingTime, setRemainingTime] = useState<moment.Duration>();
+  const [percentageComplete, setPercentageComplete] = useState<number>(0);
+
   // the beginning of the unlock cycle
   const startTime = parseTimestamp(timestamp.toString()).subtract(30, "days");
 
   // the end of the unlock cycle
   const unlockTime = parseTimestamp(timestamp.toString());
 
-  // the remaining time of the unlock cycle
-  const now = moment();
-  const remainingTime = moment.duration(
-    unlockTime.valueOf() - now.valueOf(),
-    "milliseconds",
-  );
+  useEffect(() => {
+    const interval = setInterval(() => {
+      // the remaining time of the unlock cycle
+      const now = moment();
+      const remainingTime = moment.duration(
+        unlockTime.valueOf() - now.valueOf(),
+        "milliseconds",
+      );
+      setRemainingTime(remainingTime);
 
-  // the time already passed in the unlock cycle
-  const alreadyPassedTime = moment.duration(
-    now.valueOf() - startTime.valueOf(),
-    "milliseconds",
-  );
+      // the time already passed in the unlock cycle
+      const alreadyPassedTime = moment.duration(
+        now.valueOf() - startTime.valueOf(),
+        "milliseconds",
+      );
+
+      const percentage =
+        (alreadyPassedTime.asMilliseconds() /
+          (unlockTime.valueOf() - startTime.valueOf())) *
+        100;
+      setPercentageComplete(percentage);
+    }, REFRESH_IN_MS);
+
+    const remainingTimeInMS = remainingTime?.asMilliseconds();
+
+    if (remainingTimeInMS !== undefined && remainingTimeInMS <= 0) {
+      window.location.reload();
+    }
+
+    // This represents the unmount function, in which you need to clear your interval to prevent memory leaks.
+    return () => clearInterval(interval);
+  }, [remainingTime, percentageComplete]);
```

Comment:
not completely understand why you use `remainingTime` and `percentageComplete` as a `useEffect` dependency ..? that means that this hook would be called any time the state is changes

Reply:
great catch! the dependencies i have here caused max stack depth rendering, cuz state change will trigger `useEffect` and useEffect will change state, endless loop.